### DISCUSSION TITLE
Code Scanning "Needs Setup" After Setting Up

### DISCUSSION THREAD

**Discussion by lauragonzalezzz** [community/community#42886] [Code Security] [answered]
### Select Topic Area

Bug

### Body

GitHub did not recognize it as a Code Scanning Action because my Security menu shows Code Scanning as "needs setup" and gives me a button to "Set Up Code Scanning."

My first attempt was to create a new workflow utilizing the CodeQL Code Scanning functionality. As far as I can tell, GitHub did not recognize it as a Code Scanning Action because my Security menu still shows Code Scanning as "needs setup" and gives me a button to "Set Up Code Scanning."

Next I went through the Security > Code Scanning user flow to create a new workflow which was identical to the other I'd already created and merged that into my repo, but it is still not registered as the Code Scanning workflow, and I could not identify any difference in the workflow itself or user flow (Security -> Code Scanning -> Set Up Code Scanning) which indicates that it is connected to the Security feature or not. The repo still shows that Code Scanning still needs setup.

I can't tell whether this is a bug in GitHub Actions/Security for not identifying these workflows as Code Scanning, or whether it's a bug in the user flow for building a Code Scanning workflow. Please advise!

https://github.com/microsoft/Windows-driver-samples/blob/main/.github/workflows/Code-Scanning.yml

**Comment by lauragonzalezzz** [comment, marked as answer]
Answering my own question in case someone else has this same issue.

My workflow was not registering as the Code Scanning workflow because it was not building the entire repository with which to create a baseline. If you have exceptions from the build, such as PR's with only .md or .github files/directories, you will have the same problem. I recommend removing any exceptions in order for CodeQL to scan the entire repository each time.

**Reply by vheri90** [reply]
``

**Reply by vheri90** [reply]
@vheri90
